# HG changeset patch
# User Vitaliy Filippov <vitali@st-filippov.office.custis.ru>
# Date 1297181968 -10800
Depends on: Y-000-debug-wikilog.diff

Patch includes several simple fixes required for MediaWiki to run under
more restrictive PHP 5.3.x.

Signed-off-by: Vitaliy Filippov <vitali@st-filippov.office.custis.ru>

diff -r b11befdd7427 -r 7b2ad2190a43 extensions/Interwiki/Interwiki_body.php
--- extensions/Interwiki/Interwiki_body.php
+++ extensions/Interwiki/Interwiki_body.php
@@ -176,7 +176,7 @@
 		global $wgRequest, $wgOut;
 		$prefix = $wgRequest->getVal( 'wpInterwikiPrefix' );
 		$do = $wgRequest->getVal( 'wpInterwikiAction' );
-		if( preg_match( '/[\s:&=]/', $prefix ) ) {
+		if( $do != 'delete' && preg_match( '/[\s:&=]/', $prefix ) ) {
 			$this->error( 'interwiki-badprefix', htmlspecialchars( $prefix ) );
 			$this->showForm( $do );
 			return;
@@ -198,8 +198,9 @@
 				$log->addEntry( 'iw_delete', $selfTitle, $reason, array( $prefix ) );
 			}
 			break;
+		case 'add':
+			$prefix = mb_strtolower( $prefix );
 		case 'edit':
-		case 'add':
 			$theurl = $wgRequest->getVal( 'wpInterwikiURL' );
 			$local = $wgRequest->getCheck( 'wpInterwikiLocal' ) ? 1 : 0;
 			$trans = $wgRequest->getCheck( 'wpInterwikiTrans' ) ? 1 : 0;
